app/models/cloud_network.rb

```diff
@@ -5,10 +5,45 @@ class CloudNetwork < ActiveRecord::Base
   belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "ManageIQ::Providers::CloudManager"
   belongs_to :cloud_tenant
   belongs_to :orchestration_stack
+
   has_many   :cloud_subnets, :dependent => :destroy
+  has_many   :network_ports, :dependent => :destroy
+  has_many   :floating_ips,  :dependent => :destroy
+
+  # TODO(lsmola) Defaulting CloudNetwork for Private network behaviour, otherwise I am unable to model
+  # vm.public_networks. Not specifying it here causes missing method, ans specifying CloudNetwrok::Private
+  # causes filtering networks by Provate, while I want to filter Public.
+  include CloudNetworkPrivateMixin
+
+  # TODO(lsmola) figure out what this means, like security groups used by VMs in the network? It's not being
+  # refreshed, so we can probably delete this association
   has_many   :security_groups
-  has_many   :vms
 
   # Use for virtual columns, mainly for modeling array and hash types, we get from the API
   serialize :extra_attributes
+
+  virtual_column :maximum_transmission_unit, :type => :string
+  virtual_column :port_security_enabled,     :type => :string
+
+  # Define all getters and setters for extra_attributes related virtual columns
+  %i(maximum_transmission_unit port_security_enabled).each do |action|
```

Comment:
I'd eventually like to see all of the `extra_attributes` handling code separated out ... doesn't have to be this PR